tests/CaptchaActionTest.php

```diff
@@ -28,29 +28,30 @@ protected function setUp()
      */
     protected function createController($config = [])
     {
-        return Yii::$app->controller = new Controller('test', Yii::$app, $config);
+        return $this->app->controller = new Controller('test', $this->app, $config);
     }
 
     public function testRun()
     {
         /* @var $driver Driver|\PHPUnit_Framework_MockObject_MockObject */
         $driver = $this->getMockBuilder(Driver::class)
+            ->setConstructorArgs([$this->app])
             ->setMethods(['renderImage'])
             ->getMock();
 
+
         $driver->expects($this->any())
             ->method('renderImage')
             ->willReturn('test image binary');
 
-        $action = new CaptchaAction('test', $this->createController(), [
-            'driver' => $driver
-        ]);
+        $action = new CaptchaAction('test', $this->createController());
+        $action->driver = $driver;
```

Comment:
I'm probably passing $driver the wrong way, since I needed to tweak CaptchaAction for this test to work. 
What's the accurate way to do it here?